src/compiler/sys.ts

```diff
@@ -126,6 +131,64 @@ namespace ts {
         }
     }
 
+    interface WatchedFileWithIsClosed extends WatchedFile {
+        isClosed?: boolean;
+    }
+    function pollWatchedFileQueue<T extends WatchedFileWithIsClosed>(
+        host: { getModifiedTime: NonNullable<System["getModifiedTime"]>; },
+        queue: (T | undefined)[],
+        pollIndex: number, chunkSize: number,
+        callbackOnWatchFileStat?: (watchedFile: T, pollIndex: number, fileChanged: boolean) => void
+    ) {
+        // Max visit would be all elements of the queue
+        let needsVisit = queue.length;
+        let definedValueCopyToIndex = pollIndex;
+        for (let polled = 0; polled < chunkSize && needsVisit > 0; nextPollIndex(), needsVisit--) {
+            const watchedFile = queue[pollIndex];
+            if (!watchedFile) {
+                continue;
+            }
+            else if (watchedFile.isClosed) {
+                queue[pollIndex] = undefined;
+                continue;
+            }
+
+            polled++;
+            const fileChanged = onWatchedFileStat(watchedFile, getModifiedTime(host, watchedFile.fileName));
+            if (watchedFile.isClosed) {
+                // Closed watcher as part of callback
+                queue[pollIndex] = undefined;

```

Comment:
`callbackOnWatchFileStat` can set the ` queue[pollIndex]` to undefined eg in `createDynamicPriorityPollingWatchFile` when it moves the file from one queue to other with lower priority because file has not changed for several polling sessions

Reply:
So what i meant was i will continue but we still need to check for the undefined value